napari/layers/points/points.py

```diff
@@ -427,11 +487,20 @@ def data(self, data: np.ndarray):
                         self._current_face_color, (adding, 1)
                     )
                 elif self._face_color_mode == ColorMode.CYCLE:
-                    face_color_property_value = self.current_properties[
+                    face_color_property = self.current_properties[
                         self._face_color_property
                     ][0]
+
+                    # check if the new edge color property is in the cycle map
+                    # and add it if it is not
+                    face_color_cycle_keys = [*self.face_color_cycle_map]
+                    if face_color_property not in face_color_cycle_keys:
+                        self.face_color_cycle_map[face_color_property] = next(
+                            self.face_color_cycle
+                        )
```

Comment:
As above... the amount of repeated code for face color and edge color is making me quite uncomfortable!